src/calculus.jl

```diff
@@ -0,0 +1,10 @@
+
+
+function diff(b1::Basic, b2::Basic)
```

Comment:
Yes, we want to promote everything into a Basic. That's basically what we do in C++, Python, etc.

In SymEngine we use a type code system. It's an enum which gives the type of the basic object and we can use this to determine the output.
For converting back this is useful.
```
a = Basic()
ccall(basic_add, Void, (Ptr{Basic}, Ptr{Basic}, Ptr{Basic}), &a, &b1, &b2)
t = code to get enum value
if t is Add
    return Add(a)
if t is Pow
    return Pow(a)
if t is Integer
    return Integer(a)
```
A SymEngine Integer can be converted to Julia Integers, but others can't